src/main/resources/workerScripts/fromMultiplePriorityQueues.lua

```diff
@@ -28,6 +30,7 @@ for q in queues.gmatch(queues, NEXT_QUEUE_REGEX) do
     end
 
     if payload ~= nil then
+        redis.call('LPUSH', inFlightKey, payload)
```

Comment:
Look into WorkerImpl . Right after we are getting element from the queue  (pop call inside call function ) we are providing it as a parameter to process() function . 
```

protected void poll() {
                         ----------
                        final String payload = pop(curQueue);
                        if (payload != null) {
                            process(ObjectMapperFactory.get().readValue(payload, Job.class), curQueue);
                         --------
        }
    
```

process function is wrapped by try/catch/finally block, and finally block executes   `removeInFlight(curQueue);. `
Job is executed inside process function; at the end of process function aggregation was handled already and inFlight key can be removed. 
However, if server running worker goes down unexpectedly, job successful execution can't be guaranteed; inFlight key will allow us to re-queue this job